### PR TITLE
Add API call to get a channel by its name

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Add API call to get a channel by its name

#### Ticket Link
GH: https://github.com/mattermost/platform/issues/4515
Jira: https://mattermost.atlassian.net/browse/PLT-3762

#### Checklist
- [X] Added or updated unit tests (required for all new features)
- [X] Added API documentation (required for all new APIs): https://github.com/mattermost/mattermost-api-reference/pull/51
- [X] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: api/channel.go [hunk 1]
```diff
@@ -951,6 +952,33 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
+	cchan := Srv.Store.Channel().GetByName(c.TeamId, channelname)
+
+	if cresult := <-cchan; cresult.Err != nil {
+		c.Err = cresult.Err
+		return
+	} else {
+		data := &model.Channel{}
+		data = cresult.Data.(*model.Channel)
+
```
Comment (jwilander) [inline api/channel.go]: Can you add a permission check here to make sure the user has read permission for this channel? Should be something like this:
```
if !HasPermissionToChannelContext(c, data.Id, model.PERMISSION_READ_CHANNEL) {
	return
}
```

Also can you update the unit test to test the permissions?

Comment (cpanato) [inline api/channel.go]: yes! i will do it this week!!

Comment (cpanato) [inline api/channel.go]: Done

File: api/channel.go [hunk 2]
```diff
@@ -953,6 +954,37 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	params := mux.Vars(r)
+	channelname := params["channel_name"]
+
+	cchan := Srv.Store.Channel().GetByName(c.TeamId, channelname)
+
+	if cresult := <-cchan; cresult.Err != nil {
+		c.Err = cresult.Err
+		return
+	} else {
+		data := &model.Channel{}
```
Comment (coreyhulen) [inline api/channel.go]: This line isn't needed. You can do `data := cresult.Data.(*model.Channel)`

Comment (coreyhulen) [inline api/channel.go]: Actually why isn't this just a copy of `getChannel()` that returns a `model.ChannelData` which has a `model.Channel` and `model.ChannelMember` ?

Comment (coreyhulen) [inline api/channel.go]: in my mind it should be almost identical to `getChannel`

Comment (cpanato) [inline api/channel.go]: I talked in the developers channel and asking if this api should return the members as well, was told to me for this api we just need return the channel and not the channel members.
@coreyhulen please let me know your thoughts if need return the members as well I agree with you this is similar to the getchannel

Comment (coreyhulen) [inline api/channel.go]: Sounds good, we can make adjustments in the 4.0 APIs if needed.

**Pull request merged**